engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala

```diff
@@ -210,37 +220,75 @@ trait ProgramExecutionSupport {
   private def onExecutionError(
     item: ExecutionItem,
     error: Throwable
-  )(implicit ctx: RuntimeContext): String = {
+  )(implicit ctx: RuntimeContext): Api.ExecutionError = {
     val itemName = item match {
       case ExecutionItem.Method(_, _, function) => function
       case ExecutionItem.CallData(call)         => call.getFunction.getName
     }
-    val errorMessage = getErrorMessage(error)
-    errorMessage match {
+    val executionError = getError(error)
+    executionError match {
       case Some(_) =>
         ctx.executionService.getLogger
-          .log(Level.FINE, s"Error executing a function $itemName.")
+          .log(Level.FINEST, s"Error executing a function $itemName.")
       case None =>
         ctx.executionService.getLogger
-          .log(Level.FINE, s"Error executing a function $itemName.", error)
+          .log(Level.FINEST, s"Error executing a function $itemName.", error)
     }
-    errorMessage.getOrElse(s"Error in function $itemName.")
+    executionError.getOrElse(
+      Api.ExecutionError(s"Error in function $itemName.", None)
+    )
   }
 
-  /** Get error message from throwable. */
-  private def getErrorMessage(t: Throwable): Option[String] =
+  private def getError(
+    t: Throwable
+  )(implicit ctx: RuntimeContext): Option[Api.ExecutionError] =
     t match {
-      case ex: InteropException => Some(ex.getMessage)
-      case ex: TruffleException => Some(ex.getMessage)
-      case ex: ServiceException => Some(ex.getMessage)
-      case _                    => None
+      case ex: TruffleException =>
```

Comment:
Probably worth checking if `Source.getLanguage()` is Enso and report a polyglot error in a different format.

Reply:
Also a question if we don't want to provide a stack trace to the IDE?

Reply:
Agree, a stack trace could be helpful. I'll try to add it as well.